addons/mail/models/mail_mail.py

```diff
@@ -320,7 +320,14 @@ def _send_prepare_body(self):
         """Return a specific ir_email body. The main purpose of this method
         is to be inherited to add custom content depending on some module."""
         self.ensure_one()
-        return self.body_html or ''
+
+        body = self.body_html
+
+        if hasattr(body, 'replace') and callable(getattr(body, 'replace')):
```

Comment:
I think this could be replaced by:

```suggestion
        body = self.body_html or ''
```
(and so in the return the `or ''` would not be necessary)